fdk/context.py

```diff
@@ -138,27 +245,63 @@ def context_from_format(format_def: str, **kwargs) -> (
 
     app_id = os.environ.get(constants.FN_APP_ID)
     fn_id = os.environ.get(constants.FN_ID)
-
+    app_name = os.environ.get(constants.FN_APP_NAME)
+    fn_name = os.environ.get(constants.FN_NAME)
+    # the tracing enabled env variable is passed as a "0" or "1" string and therefore needs to be converted appropriately. 
+    is_tracing_enabled = os.environ.get(constants.OCI_TRACING_ENABLED)
+    is_tracing_enabled = bool(int(is_tracing_enabled)) if is_tracing_enabled != None else False
+    trace_collector_url = os.environ.get(constants.OCI_TRACE_COLLECTOR_URL)
+    
     if format_def == constants.HTTPSTREAM:
         data = kwargs.get("data")
         headers = kwargs.get("headers")
 
+        # zipkin tracing http headers
+        trace_id = span_id = parent_span_id = is_sampled = trace_flags = tracing_context = None
+        if is_tracing_enabled: 
+          # we generate the trace_id if tracing is enabled but the traceId zipkin header is missing. 
+          trace_id = headers.get(constants.X_B3_TRACEID)
+          trace_id = generate_id() if trace_id == None else trace_id
+          
+          # As the fnLb sends the parent_span_id as the span_id, we must account for this.
+          span_id = headers.get(constants.X_B3_SPANID)
+          parent_span_id = headers.get(constants.X_B3_PARENTSPANID)
+          if parent_span_id == None and span_id != None:
+            parent_span_id = span_id
+            span_id = generate_id()
+
+          # span_id is also generated if the zipkin header is missing.
+          span_id = generate_id() if span_id == None else span_id
```

Comment:
I guess this is safe and better to get a span emitted for debugging if something goes wrong with the downstream service.